File: chapter-02-tiny-screen-interaction/wearOS/WristCounterList.kt
package com.example.handsonwearables.chapter02

import androidx.compose.runtime.Composable
import androidx.compose.runtime.getValue
import androidx.compose.runtime.mutableIntStateOf
import androidx.compose.runtime.remember
import androidx.compose.runtime.setValue
import androidx.wear.compose.foundation.lazy.ScalingLazyColumn
import androidx.wear.compose.foundation.lazy.items
import androidx.wear.compose.material3.Button
import androidx.wear.compose.material3.Text

// Complete file around the book's p.28 ScalingLazyColumn snippet.
@Composable
fun WristCounterList() {
    var count by remember { mutableIntStateOf(0) }
    val history = remember(count) { (1..count).map { "Count reached $it" } }

    ScalingLazyColumn {
        item { Text("Count: $count") }
        item { Button(onClick = { count++ }) { Text("+1") } }
        items(history) { entry ->
            Text(entry)
        }
    }
}

File: chapter-02-tiny-screen-interaction/watchOS/WristCounterList.swift
import SwiftUI

// Complete view around the book's p.29 List snippet.
struct WristCounterList: View {
    @State private var count = 0
    private var history: [String] { (1...max(count, 1)).map { "Count reached \($0)" } }

    var body: some View {
        List {
            Section("Count: \(count)") {
                Button("+1") { count += 1 }
                Button("Reset") { count = 0 }
            }
            Section("History") {
                ForEach(history, id: \.self) { entry in
                    Text(entry)
                }
            }
        }
    }
}
